src/runmode-dpdk.c

```diff
@@ -0,0 +1,1274 @@
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetCopyIfaceSettings(iconf, copy_iface_str, copy_mode_str);
+    if (retval < 0) {
+        return retval;
+    }
+
+    SCReturnInt(0);
+}
+DPDKIfaceConfig *ConfigParse(const char *iface)
+{
+    int retval;
+    DPDKIfaceConfig *iconf = NULL;
+    if (iface == NULL) {
+        SCLogError(SC_ERR_DPDK_CONF, "Provided iface is NULL");
+        SCReturnPtr(NULL, "void *");
+    }
+
+    ConfigInit(&iconf);
+    if (iconf == NULL) {
+        SCReturnPtr(NULL, "void *");
+    }
+    retval = ConfigLoad(iconf, iface);
+    if (retval < 0) {
+        iconf->DerefFunc(iconf);
+        SCReturnPtr(NULL, "void *");
+    }
+
+    return iconf;
+}
+
+void DeviceInitPortConf(const DPDKIfaceConfig *iconf, const struct rte_eth_dev_info *dev_info,
+        struct rte_eth_conf *port_conf)
+{
+    *port_conf = (struct rte_eth_conf){
+            .rxmode = {
+                    .mq_mode = ETH_MQ_RX_NONE,
+                    .max_rx_pkt_len = RTE_ETHER_MAX_LEN,
+                    .offloads = 0, // turn every offload off to prevent any packet modification
+            },
+            .txmode = {
+                    .mq_mode = ETH_MQ_TX_NONE,
+                    .offloads = 0,
+            },
+    };
+
+    // configure RX offloads
+    if (dev_info->rx_offload_capa & DEV_RX_OFFLOAD_RSS_HASH) {
+        if (iconf->nb_rx_queues > 1) {
+            SCLogConfig("RSS enabled on %s for %d queues", iconf->iface, iconf->nb_rx_queues);
+            port_conf->rx_adv_conf.rss_conf.rss_key = rss_hkey;
+            port_conf->rx_adv_conf.rss_conf.rss_key_len = RSS_HKEY_LEN;
+            port_conf->rx_adv_conf.rss_conf.rss_hf = ETH_RSS_TCP | ETH_RSS_UDP;
+            port_conf->rxmode.mq_mode = ETH_MQ_RX_RSS;
+        } else {
+            SCLogConfig("RSS not enabled on %s", iconf->iface);
+            port_conf->rx_adv_conf.rss_conf.rss_key = NULL;
+            port_conf->rx_adv_conf.rss_conf.rss_hf = 0;
+        }
+    } else {
+        SCLogConfig("RSS not supported on %s", iconf->iface);
+    }
+
+    if (iconf->checksum_mode == CHECKSUM_VALIDATION_DISABLE) {
+        SCLogConfig("Checksum validation disabled on %s", iconf->iface);
+    } else if (dev_info->rx_offload_capa & DEV_RX_OFFLOAD_CHECKSUM) {
+        if (iconf->checksum_mode == CHECKSUM_VALIDATION_ENABLE &&
+                iconf->flags & DPDK_RX_CHECKSUM_OFFLOAD) {
+            SCLogConfig("Checksum validation enabled and offloaded on %s", iconf->iface);
+            port_conf->rxmode.offloads |= DEV_RX_OFFLOAD_CHECKSUM;
+        } else if (iconf->checksum_mode == CHECKSUM_VALIDATION_ENABLE &&
+                   !(iconf->flags & DPDK_RX_CHECKSUM_OFFLOAD)) {
+            SCLogConfig("Checksum validation enabled but not offloaded on %s", iconf->iface);
+        }
+    }
+
+    // configure TX offloads
+    if (dev_info->tx_offload_capa & DEV_TX_OFFLOAD_MT_LOCKFREE) {
+        port_conf->txmode.offloads |= DEV_TX_OFFLOAD_MT_LOCKFREE;
```

Comment:
as far I recollect `DEV_TX_OFFLOAD_MT_LOCKFREE` is not supported by all DPDK listed NIC. This limits the operation to a specific vendor NIC.  rework

Reply:
please check DPDK documentation for NIC feature support. You can easily understand, Also you check-in DPODK API documentation where it is called out specifically.

Reply:
I have gone through the documentation but I can see no problem with this code. I am not using any fields of the mbuf structure, I only validate the support of NIC functionality. Therefore, I believe this approach is vendor-independent. The DPDK API documentation only mentioned that if an offload is supported then the mbuf structure can be "enriched" by extra fields. I do not use any non-standard fields in the code. The DPDK documentation does not mention that you can not verify the capabilities of the NIC by using `dev_info->tx_offload_capa`. 

However, I realized that I do not use the offload `DEV_TX_OFFLOAD_MT_LOCKFREE`, so I have removed its enablement.